generators/JSONtoTests.jl

```diff
@@ -0,0 +1,138 @@
+# This code generates the runtets.jl file automatically for some lucky exercises that work with this script.
+
+# NOTE: This is a very experimental version of this idea. The code is not all that well written,
+# and may not work on most exercises. If it works, it might save you hours of time writing that exercise.
+# If it doesn't it will throw an error and quit.
+
+using JSON, HTTP
+
+# Convert an array to a array in string form
+# eg. 4-element Array{Int64,1}: 1 2 3 4 --> "[1, 2, 3, 4]"
+function array_to_str(array)
+    # Start array with opening bracket
+    str = ["["]
+    # If array is empty, return an empty array
+    if length(array) == 0
+        return "[]"
+    else
+    # Else, for each item in array, add it to array
+        for item in array
+            # If item is a string, enclose it in quotations and add it to our array
+            if typeof(item) == String
+                push!(str, "\"$item\", ")
+            # If item is an array, call function recursively to add it to our array
+            elseif typeof(item) == AbstractArray
+                push!(str, array_to_str(item))
+            # Else, if item is something else, just write it to array
+            else                
+                push!(str, "$item, ")
+            end
+        end
+
+        # Make it a string and remove the last extra comma and space from array
+        truncated_str = join(str)[1:end-2]
+
+        # Return final array string by closing with bracket
+        return string(truncated_str, "]")
+    end
```

Comment:
No need to wrap it in an `else` block due to the `return` in L16.

Reply:
That's for the non-empty arrays. The return on L16 is a special case handle for an empty array.

Reply:
Yes but unless I'm misunderstanding the code, there are two cases here: 

1. The condition of the `if` statement is true, then it will return immediately and everything below that won't be executed.
2. It's false, the entire `else` block will be executed.

Everything below is wrapped in the else block, so you could leave out the else block. Keep the if and return above it, but don't wrap the entire rest inside the else block. This saves an indentation level and is a bit more readable. This is often referred to as "aligning the happy path to the left" or something similar.

```julia
function array_to_str(array)
    # Start array with opening bracket
    str = ["["]
    # If array is empty, return an empty array
    if length(array) == 0
        return "[]"
    end

    # Else, for each item in array, add it to array
    for item in array
        # If item is a string, enclose it in quotations and add it to our array
        if typeof(item) == String
            push!(str, "\"$item\", ")
        # If item is an array, call function recursively to add it to our array
        elseif typeof(item) == AbstractArray
            push!(str, array_to_str(item))
        # Else, if item is something else, just write it to array
        else                
            push!(str, "$item, ")
        end
    end

    # Make it a string and remove the last extra comma and space from array
    truncated_str = join(str)[1:end-2]

    # Return final array string by closing with bracket
    return string(truncated_str, "]")
end
```

It's also possible that I'm misunderstanding or overreading something, but I think this should work.

Reply:
Aha! I see what you mean now. I’ll add this too in the next commit.